test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -90,11 +90,12 @@ public void setup() {
         assertTrue(segmentStoreInstance.isRunning());
         log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
 
-        //executor service
-        executorService = Executors.newScheduledThreadPool(NUM_READERS + NUM_WRITERS);
+        //num. of readers + num. of writers + 1 to run checkScale operation
+        executorService = Executors.newScheduledThreadPool(NUM_READERS + NUM_WRITERS + 1);
```

Comment:
so the conjuncture is that readers and writers busy loop on their threads without relinquishing it ever to the pool?